### PR TITLE
config: Add detail to auto_join_rooms comment

### PR DESCRIPTION
**Pull request by dykstranet**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: synapse/config/registration.py [hunk 1]
```diff
@@ -378,6 +378,9 @@ def generate_config_section(self, generate_secrets=False, **kwargs):
         # By default, any room aliases included in this list will be created
         # as a publicly joinable room when the first user registers for the
         # homeserver. This behaviour can be customised with the settings below.
+        # If the room already exists, make certain it is a publicly joinable
+        # room, i.e. 'Security & Privacy' settings for the room allow 'Anyone
+        # who knows the room's link, including guests' to access the room.
```
Comment (anoadragon453) [inline synapse/config/registration.py:383]: This is very helpful, though I'm afraid Element is only one client in the Matrix ecosystem, and we try to be client-agnostic in Synapse.

The requirement for the room is to be publicly joinable (have a join rule of `public` set). Additionally one doesn't need to allow guest access to the room.

```suggestion
        # room. The join rule of the room must be set to 'public'.
```

While I'd like to make this as easy as possible for people to use, client-specific instructions are best suited to a client-specific guide.

File: changelog.d/9291.doc [hunk 2]
```diff
@@ -0,0 +1 @@
+Add configuration detail to auto_join_rooms comment.
```
Comment (anoadragon453) [inline changelog.d/9291.doc:1]:
```suggestion
Add note to `auto_join_rooms` config option explaining existing rooms must be publicly joinable.
```

**Review by anoadragon453 — APPROVED** [review]
Important gotcha to call out, thanks!

Test failures were unrelated to the contents of this PR.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
That was meant to be a change request :)

**Comment by dykstranet** [comment]
@anoadragon453 Thank you for your suggestion, I have applied these changes. Feel free to squash the PR.

**Review by anoadragon453 — APPROVED** [review]
lgtm, thank you!

**Pull request merged**